### DISCUSSION TITLE
Running just the scheduled worker

### DISCUSSION THREAD

**Discussion by jamietanna** [mend/renovate-ce-ee#284] [Q&A]
Related to #104, one option we're looking at to scale out the Worker is to schedule executions of the Worker process, for a subset of repos, on a given schedule.

This will run alongside the Renovate On Prem persistent service (for webhook functionality).

As noted in https://docs.renovatebot.com/examples/self-hosting/#kubernetes there's the ability to run the Open Source Renovate on a schedule - is it possible to configure the Renovate On-Prem product like so? Or would it be best to have a mix of the two?

**Comment by rarkins** [comment]
Renovate On-Prem is a long lived server and not a CLI like Renovate OSS, so the concepts aren't 1:1. Renovate On-Prem has a scheduler with cron syntax and this controls *when jobs are enqueued*. If your queue is really long then you can end up with jobs running around the clock, essentially independent of the cron schedule.

**Comment by jamietanna** [comment]
Gotcha 👍🏽

For jobs that are `enqueued` (ie through schedule or by the HEAD of the repo being updated) the work is the same - rescan the repos dependencies?

In which case for a repo that's updated daily, rescanning dependencies may be less important, so we could reduce the frequency of the scheduled scans?

**Comment by rarkins** [comment]
There are multiple reasons why webhooks trigger a job, including:
- Closing or merging of a Renovate PR
- Default branch commit updating either a known package file or config file
- Users ticking rebase/retry

In such cases those "on demand" jobs take higher priority in the job queue.

You could set scheduling to be as infrequent as you like and rely on the dynamic jobs but remember it doesn't apply to *every* commit
